docs/hub/datasets-spark.md

```diff
@@ -0,0 +1,441 @@
+
+For example you can read Parquet files from Hugging Face in an optimized way using PyArrow using this `read_parquet` helper function:
+
+```python
+from functools import partial
+from typing import Iterator, Optional, Union
+
+import pyarrow as pa
+import pyarrow.parquet as pq
+from huggingface_hub import HfFileSystem
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.pandas.types import from_arrow_schema
+
+
+def _read(iterator: Iterator[pa.RecordBatch], **kwargs) -> Iterator[pa.RecordBatch]:
+    for batch in iterator:
+        for path in batch[0].to_pylist():
+            yield from pq.read_table(path, **kwargs).to_batches()
+
+
+def read_parquet(
+    path: str,
+    columns: Optional[list[str]] = None,
+    filters: Optional[Union[list[tuple], list[list[tuple]]]] = None,
+    **kwargs,
+) -> DataFrame:
+    """
+    Loads Parquet files from Hugging Face using PyArrow, returning a PySPark `DataFrame`.
+
+    It reads Parquet files in a distributed manner.
+
+    Access private or gated repositories using `huggingface-cli login` or passing a token
+    using the `storage_options` argument: `storage_options={"token": "hf_xxx"}`
+
+    Parameters
+    ----------
+    path : str
+        Path to the file. Prefix with a protocol like `hf://` to read from Hugging Face.
+        You can read from multiple files if you pass a globstring.
+    columns : list, default None
+        If not None, only these columns will be read from the file.
+    filters : List[Tuple] or List[List[Tuple]], default None
+        To filter out data.
+        Filter syntax: [[(column, op, val), ...],...]
+        where op is [==, =, >, >=, <, <=, !=, in, not in]
+        The innermost tuples are transposed into a set of filters applied
+        through an `AND` operation.
+        The outer list combines these sets of filters through an `OR`
+        operation.
+        A single list of tuples can also be used, meaning that no `OR`
+        operation between set of filters is to be conducted.
+
+    **kwargs
+        Any additional kwargs are passed to pyarrow.parquet.read_table.
+
+    Returns
+    -------
+    DataFrame
+        DataFrame based on parquet file.
+
+    Examples
+    --------
+    >>> path = "hf://datasets/username/dataset/data.parquet"
+    >>> pd.DataFrame({"foo": range(5), "bar": range(5, 10)}).to_parquet(path)
+    >>> read_parquet(path).show()
+    +---+---+
+    |foo|bar|
+    +---+---+
+    |  0|  5|
+    |  1|  6|
+    |  2|  7|
+    |  3|  8|
+    |  4|  9|
+    +---+---+
+    >>> read_parquet(path, columns=["bar"]).show()
+    +---+
+    |bar|
+    +---+
+    |  5|
+    |  6|
+    |  7|
+    |  8|
+    |  9|
+    +---+
+    >>> sel = [("foo", ">", 2)]
+    >>> read_parquet(path, filters=sel).show()
+    +---+---+
+    |foo|bar|
+    +---+---+
+    |  3|  8|
+    |  4|  9|
+    +---+---+
+    """
+    filesystem: HfFileSystem = kwargs.pop("filesystem", HfFileSystem(**kwargs.pop("storage_options", {})))
```

Comment:
```suggestion
    filesystem: HfFileSystem = kwargs.pop("filesystem") if "filesystem" in kwargs else HfFileSystem(**kwargs.pop("storage_options", {}))
```

(nit) doing so avoid to instantiate a `HfFileSystem` obj for nothing if `filesystem` already defined.